src/Features/Core/Portable/UseNameOf/AbstractUseNameOfDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,101 @@
+﻿namespace Microsoft.CodeAnalysis.UseNameof
+{
+    using System.Collections.Immutable;
+    using System.Linq;
+    using System.Threading;
+    using Microsoft.CodeAnalysis.CodeStyle;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using Microsoft.CodeAnalysis.Editing;
+    using Microsoft.CodeAnalysis.LanguageServices;
+    using Microsoft.CodeAnalysis.Operations;
+
+    internal abstract class AbstractUseNameOfDiagnosticAnalyzer : AbstractCodeStyleDiagnosticAnalyzer
+    {
+        protected AbstractUseNameOfDiagnosticAnalyzer()
+            : base(
+                IDEDiagnosticIds.UseNameofDiagnosticId,
+                "Use nameof.",
+                "Use nameof.",
+                false)
+        {
+        }
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory() => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
+
+        public override bool OpenFileOnly(Workspace workspace) => false;
+
+        protected abstract ISyntaxFactsService GetSyntaxFactsService();
+
+        protected override void InitializeWorker(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.RegisterOperationAction(Handle, OperationKind.Literal);
+        }
+
+        private void Handle(OperationAnalysisContext context)
+        {
+            if (context.Operation is ILiteralOperation literal &&
+                literal.Parent is IArgumentOperation &&
+                literal.Type.SpecialType == SpecialType.System_String &&
+                literal.ConstantValue.HasValue &&
+                literal.ConstantValue.Value is string value &&
+                GetSyntaxFactsService() is ISyntaxFactsService syntaxFacts &&
+                syntaxFacts.IsValidIdentifier(value))
+            {
+                var symbol = context.Compilation.GetSemanticModel(literal.Syntax.SyntaxTree)
+                                    .LookupSymbols(literal.Syntax.SpanStart, name: value)
+                                    .FirstOrDefault();
```

Comment:
The above is probably very suboptimal, I could not find API for getting a `SemanticModel` in `OperationAnalysisContext`
Also lookup followed by filtering is not ideal, wrote it as is because I'm curious about if there is a cleaner way.
If not I'll change it to:
1. Check parameters for containing symbol if `IMethodSymbol`
2. Check members for containing type and super types.
3. Use a walker to walk locals, or skip locals altogether.

Reply:
@mavasani is htere a reason that OperationAnalysisContext does not expose te semantic model?  looking at where it is created, it seems to have a semantic model readily available

Reply:
We initially exposed this in the IOperation prototype, but John removed it given it could be null for MSIL to IOperation raise scenarios. I think we should still expose it with a doc comment about this scenario. I believe we had an issue tracking this work, but I cannot find it currently. @JohanLarsson, can you please file an issue and assign it to me?

Reply:
https://github.com/dotnet/roslyn/issues/26180
Don't know how to assign.